torch/_inductor/template_heuristics/cutedsl.py

```diff
@@ -0,0 +1,293 @@
+        CuTeGemmConfig(
+            TILE_M=128,
+            TILE_N=256,
+            CLUSTER_M=1,
+            CLUSTER_N=2,
+            USE_2_CTA=False,
+            TENSORMAP_UPDATE_MODE=TensorMapUpdateMode.GMEM,
+        ),
+        CuTeGemmConfig(
+            TILE_M=64,
+            TILE_N=32,
+            CLUSTER_M=1,
+            CLUSTER_N=1,
+            USE_2_CTA=False,
+            TENSORMAP_UPDATE_MODE=TensorMapUpdateMode.SMEM,
+        ),
+        CuTeGemmConfig(
+            TILE_M=256,
+            TILE_N=256,
+            CLUSTER_M=2,
+            CLUSTER_N=1,
+            USE_2_CTA=True,
+            TENSORMAP_UPDATE_MODE=TensorMapUpdateMode.SMEM,
+        ),
+        CuTeGemmConfig(
+            TILE_M=128,
+            TILE_N=256,
+            CLUSTER_M=1,
+            CLUSTER_N=1,
+            USE_2_CTA=False,
+            TENSORMAP_UPDATE_MODE=TensorMapUpdateMode.GMEM,
+        ),
+        CuTeGemmConfig(
+            TILE_M=256,
+            TILE_N=256,
+            CLUSTER_M=8,
+            CLUSTER_N=1,
+            USE_2_CTA=True,
+            TENSORMAP_UPDATE_MODE=TensorMapUpdateMode.GMEM,
+        ),
+        CuTeGemmConfig(
+            TILE_M=64,
+            TILE_N=32,
+            CLUSTER_M=1,
+            CLUSTER_N=2,
+            USE_2_CTA=False,
+            TENSORMAP_UPDATE_MODE=TensorMapUpdateMode.SMEM,
+        ),
+        CuTeGemmConfig(
+            TILE_M=256,
+            TILE_N=192,
+            CLUSTER_M=2,
+            CLUSTER_N=1,
+            USE_2_CTA=True,
+            TENSORMAP_UPDATE_MODE=TensorMapUpdateMode.GMEM,
+        ),
+        CuTeGemmConfig(
+            TILE_M=256,
+            TILE_N=256,
+            CLUSTER_M=2,
+            CLUSTER_N=2,
+            USE_2_CTA=True,
+            TENSORMAP_UPDATE_MODE=TensorMapUpdateMode.SMEM,
+        ),
+        CuTeGemmConfig(
+            TILE_M=128,
+            TILE_N=96,
+            CLUSTER_M=1,
+            CLUSTER_N=2,
+            USE_2_CTA=False,
+            TENSORMAP_UPDATE_MODE=TensorMapUpdateMode.SMEM,
+        ),
+        CuTeGemmConfig(
+            TILE_M=64,
+            TILE_N=192,
+            CLUSTER_M=1,
+            CLUSTER_N=1,
+            USE_2_CTA=False,
+            TENSORMAP_UPDATE_MODE=TensorMapUpdateMode.SMEM,
+        ),
+        CuTeGemmConfig(
+            TILE_M=64,
+            TILE_N=64,
+            CLUSTER_M=1,
+            CLUSTER_N=1,
+            USE_2_CTA=False,
+            TENSORMAP_UPDATE_MODE=TensorMapUpdateMode.GMEM,
+        ),
+        CuTeGemmConfig(
+            TILE_M=64,
+            TILE_N=192,
+            CLUSTER_M=1,
+            CLUSTER_N=1,
+            USE_2_CTA=False,
+            TENSORMAP_UPDATE_MODE=TensorMapUpdateMode.GMEM,
+        ),
+        CuTeGemmConfig(
+            TILE_M=128,
+            TILE_N=64,
+            CLUSTER_M=1,
+            CLUSTER_N=1,
+            USE_2_CTA=False,
+            TENSORMAP_UPDATE_MODE=TensorMapUpdateMode.GMEM,
+        ),
+        CuTeGemmConfig(
```

Comment:
nit: for cleaning up the code a bit can you make these a list of tuples and then apply the constructor to the unpacked tuples? e.g. ([CuTeGemmConfig(*args) for args in configs]) and if there's a pattern can you autogenerate?